### DISCUSSION TITLE
Xcode Key Bindings

### DISCUSSION THREAD

**Discussion by KieranConlon** [intitni/CopilotForXcode#103] [Ideas]
With the addition of the new GPT chat feature, I have been trying out some Xcode key bindings to access this feature.

I did add my [suggestions](https://github.com/intitni/CopilotForXcode/issues/14#issuecomment-1484860426) to the related issue, but being a closed issue, I don't suppose it will be seen - so just re-posting here.

I'm using a English-UK keyboard layout, so these bindings probably don't work for everyone; what key bindings are you using?

---

New key bindings for the GPT chat feature:
CoPilot -> Chat with Selection: ⌥"
CoPilot -> Explain Selection: ⌥|

Again, using ⌥⇧ as the "access" key combination for these bindings.

The rationale for these bindings...
**Chat**:  Using the 'quote' symbol for starting an OpenAI chat session has an analogy to starting a conversation with OpenAI where the initial question is about the selected (or quoted) text.
**Explain**:  Using the 'pipe' symbol for starting an OpenAI session to explain the selected code has an analogy with command line tools (e.g. grep) for 'piping' text to another command so that additional processing of the code/selection can happen.

![image](https://user-images.githubusercontent.com/33759139/227910263-d6076d01-bb3d-4638-b3b6-f87fa80ad6fe.png)

These bindings do not seem to conflict with standard key bindings used on Xcode.

![image](https://user-images.githubusercontent.com/33759139/227909229-c487b41a-bc4f-46ca-9dc6-95760dbc09f2.png)

**Comment by shanecowherd** [comment]
This helped, thank you @KieranConlon
